### ISSUE TITLE
Not able to disable 'Lint/UnneededRequireStatement'

### ISSUE THREAD

**Issue by swapab** [rubocop-hq/rubocop#4940]
If I disable `Lint/UnneededRequireStatement` with
```
require 'thread' # rubocop:disable Lint/UnneededRequireStatement
...
```

Rubocop warns about offense saying `Unnecessary disabling of Lint/UnneededRequireStatement`:
```
Offenses:
lib/foreman_maintain/reporter/cli_reporter.rb:1:18: W: Unnecessary disabling of Lint/UnneededRequireStatement.
require 'thread' # rubocop:disable Lint/UnneededRequireStatement
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
157 files inspected, 1 offense detected
RuboCop failed!

```
---

Observed this only on Travis-CI (as of now)
https://travis-ci.org/theforeman/foreman_maintain/jobs/293000847
https://travis-ci.org/theforeman/foreman_maintain/jobs/293001743

*Ruby version on Travis-CI is 2.3.3*

My local env. is not reporting any such warning.

## RuboCop version
From my local development environment
```
$ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.2.0 x86_64-linux)
```

PS: I need `require 'thread'` to support ruby 1.8.7

**Comment by pocke** [comment]
`Lint/UnneededRequireStatement` is not available if `TargetRubyVersion` is not configured. Because the cop is available on Ruby 2.2 or later. https://github.com/bbatsov/rubocop/blob/dd57e95e27ce70c9938ba480575ffbd61dbe56a8/lib/rubocop/cop/lint/unneeded_require_statement.rb#L27

So, if you need to enable the cop, please add `TargetRubyVersion` to your `.rubocop.yml`. But it sometimes doesn't work well because you need to support ruby 1.8.7.
Or remove the disable comment, and check your `.rubocop.yml` in your home directory(maybe it has `TargetRubyVersion`).


reproduce
===

Without `.rubocop.yml`, with the disable comment
---

```bash
$ rubocop
Inspecting 1 file
W

Offenses:

test.rb:1:18: W: Unnecessary disabling of Lint/UnneededRequireStatement.
require 'thread' # rubocop:disable Lint/UnneededRequireStatement
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```


With `.rubocop.yml`, with the disable comment
------


```yaml
AllCops:
  TargetRubyVersion: 2.2
```

```bash
$ rubocop
Inspecting 1 file
.

1 file inspected, no offenses detected
```

**Comment by bbatsov** [comment]
> PS: I need require 'thread' to support ruby 1.8.7

Not really on topic, but I'm still amazed after all those years 1.8 and 1.9 refuse to die. :-)

**Comment by gvieira** [comment]
@bbatsov Just out of curiosity, I have a huge Ruby project that needs to be run on Windows machines with some specific DB gems (for integration stuff) that still depends on 1.9. I cry a little every time I need to install Ruby 1.9.3.

**Comment by koic** [comment]
I think that this issue will be resolved by @pocke 's comment.
https://github.com/rubocop-hq/rubocop/issues/4940#issuecomment-339838985

If you have a problem using the latest RuboCop, please reopen and comment. Thank you.
